Free PQescapeIdentifier/PQescapeIdentifier after used to avoid memory leak

harry-leq opened the pull request:
- PQescapeIdentifier/PQescapeIdentifier use malloc to allocate, which requires PQfreemem to deallocate after used.
- Pointers are set to NULL to avoid being dangling in any cases.

Commit:
Free PQescapeIdentifier after used to avoid memory leak due to malloc

Commit:
Free PQescapeLiteral after used to avoid memory leak due to malloc

nmisch commented on pglogical_create_subscriber.c:
```diff
@@ -1040,9 +1076,15 @@ initialize_replication_origin(PGconn *conn, char *origin_name, char *remote_lsn)
-						  PQescapeLiteral(conn, origin_name, strlen(origin_name)),
-						  remote_lsn ? PQescapeLiteral(conn, remote_lsn, strlen(remote_lsn)) : "0");
+						  escaped_origin_name,
+						  remote_lsn ? escaped_remote_lsn : "0");
```
If remote_lsn==NULL, we'll have crashed at the earlier strlen(remote_lsn).  Would you account for that?

harry-leq replied:
I have push a new commit to fix this issue.

Commit:
initialize_replication_origin: only set escaped_remote_lsn when remote_lsn is not null

initialize_replication_origin: only set escaped_remote_lsn when remote_lsn is not null

nmisch commented on pglogical_create_subscriber.c:
```diff
@@ -801,17 +801,29 @@ initialize_replication_slot(PGconn *conn, char *dbname,
+	escaped_subscription_name = PQescapeLiteral(conn, subscription_name, strlen(subscription_name));
 	printfPQExpBuffer(&query,
 					  "SELECT pglogical.pglogical_gen_slot_name(%s, %s, %s)",
-					  PQescapeLiteral(conn, dbname, strlen(dbname)),
-					  PQescapeLiteral(conn, provider_node_name,
-									  strlen(provider_node_name)),
-					  PQescapeLiteral(conn, subscription_name,
-									  strlen(subscription_name)));
+					  escaped_dbname,
+					  escaped_provider_node_name,
+					  escaped_subscription_name);
```
True.  Ideally we'd have wrappers that ereport(ERROR) if these return NULL.  That flaw is preexisting and orthogonal to this PR, however.  Likewise some of the other review comments below.

nmisch reviewed (changes requested):
Thanks for your patience.  Please follow local formatting conventions per copilot's comments, and remove whitespace that `git diff --check '@{u}'` finds.  Then I'll merge this.

nmisch commented on pglogical_create_subscriber.c:
```diff
@@ -1040,9 +1076,17 @@ initialize_replication_origin(PGconn *conn, char *origin_name, char *remote_lsn)
-						  PQescapeLiteral(conn, origin_name, strlen(origin_name)),
-						  remote_lsn ? PQescapeLiteral(conn, remote_lsn, strlen(remote_lsn)) : "0");
+						  escaped_origin_name,
+						  escaped_remote_lsn ? escaped_remote_lsn : "0");
```
If PQescapeLiteral() returns NULL due to ENOMEM, this will now use "0" instead
of a NULL pointer.  A segfault is better than passing an irrelevant "0", so
restore the old behavior of that scenario.

harry-leq replied:
I restored the logic back to `remote_lsn ? escaped_remote_lsn : "0"` to produce segfault for ENOMEM in escaped_remote_lsn

Commit:
nit: correct spelling in log message
